### PR TITLE
ArmoredChat Fixes

### PR DESCRIPTION
**Pull request by Armored-Dragon**
Finalization of the Armored Chat application

### CODE REVIEW

File: applications/armored-chat/armored_chat.js [hunk 1]
```diff
@@ -135,7 +136,8 @@
         break;
 
       case "open_url":
-        Window.openUrl(parsed.message.toString());
+        new OverlayWebWindow({ source: parsed.url.toString(), width: 500, height: 400 });
+        // Window.openUrl(parsed.url.toString());
```
Comment (daleglass) [inline applications/armored-chat/armored_chat.js:140]: Commented out code

File: applications/armored-chat/index.html [hunk 2]
```diff
@@ -186,30 +186,34 @@
...
     if (yt_url) {
-      message.message = message.message.replace(
-        yt_url,
-        `${yt_url}
-          <br>
-          <iframe class="z-depth-2" width='420' height='236' src='https://www.youtube.com/embed/${yt_url.toString().split("/")[3]}' frameborder='0'></iframe>`
-      );
+      yt_url.forEach((url) => {
+        message_embeds += `<iframe class="z-depth-2" width='420' height='236' src='https://www.youtube.com/embed/${url.toString().split("/")[3]}' frameborder='0'></iframe><br>`;
+        // message.message = message.message.replace(url, "");
```
Comment (daleglass) [inline applications/armored-chat/index.html:198]: Commented out code

File: applications/armored-chat/index.html [hunk 3]
```diff
@@ -186,30 +186,34 @@
...
+      });
     }
 
     // Image embeds
     let image_link = message.message.match(/.+.(png|jpg|jpeg|webp)/g);
     if (image_link) {
-      message.message = message.message.replace(image_link, `${image_link}<br><span class='image-container'><img src='${image_link}'></span>`);
+      image_link.forEach((image) => {
+        message_embeds += `<span class='image-container'><img src='${image}'></span><br>`;
+        // message.message = message.message.replace(image, "");
```
Comment (daleglass) [inline applications/armored-chat/index.html:207]: And here

File: applications/armored-chat/index.scss [hunk 4]
```diff
@@ -104,8 +108,33 @@ body {
...
+            max-height: 300px;
+
+            img {
+              // width: auto;
+              // height: 100%;
               max-width: 400px;
               max-height: 300px;
+
+              // max-width: 400px;
+              // max-height: 300px;
```
Comment (daleglass) [inline applications/armored-chat/index.scss:137]: And here.

Also, are px appropriate? What about 4k screens?

**Review by daleglass — CHANGES_REQUESTED** [review]

File: applications/armored-chat/index.scss (hunk 4 above)
Comment (Armored-Dragon) [reply applications/armored-chat/index.scss:137]: The stylesheet is capping the preview image size and not the full size. You can see the full size images when you click on the embed link directly.

**Pull request merged**